maas/resource_maas_network_interface_vlan.go

```diff
@@ -0,0 +1,226 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/maas/gomaasclient/client"
+	"github.com/maas/gomaasclient/entity"
+)
+
+func resourceMaasNetworkInterfaceVlan() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS network Vlans.",
+		CreateContext: resourceMaasNetworkInterfaceVlanCreate,
+		ReadContext:   resourceMaasNetworkInterfaceVlanRead,
+		UpdateContext: resourceMaasNetworkInterfaceVlanUpdate,
+		DeleteContext: resourceMaasNetworkInterfaceVlanDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMaasNetworkInterfaceVlanImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"accept_ra": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				Description: "Accept router advertisements. (IPv6 only).",
+			},
+			"fabric": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The identifier (name or ID) of the fabric for the new VLAN.",
+			},
+			"machine": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "List of MAAS machines' identifiers (system ID, hostname, or FQDN) that will be tagged with the new tag.",
+			},
+			"mtu": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Maximum transmission unit.",
+			},
+			"parent": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Parent interface name for this bridge interface.",
+			},
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Tags for the interface.",
+			},
+			"vlan": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "VLAN the interface is connected to.",
+			},
+		},
+	}
+}
+
+func resourceMaasNetworkInterfaceVlanCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Same as bond and bridge, please remove the `Maas` part of every function except `resourceMaasNetworkInterfaceVlan`